notification/notification.js

```diff
@@ -1,40 +1,189 @@
-// on swapping tab update URL
-function updateCurrentUrl() {
-	function logTabs(tabs) {
-		const tab = tabs[0]; // Safe to assume there will only be one result
-		showNotification(tab.url);
-	}
-	function onError(err){
-		showNotification(null);
-	}
-	browser.tabs.query({currentWindow: true, active: true}).then(logTabs, onError);
-}
+﻿﻿function updateCurrentUrl() {
+    function logTabs(tabs) {
+        if(tabs[0].url) {
+            handleMessage(tab[0].url);
+        }
+     }
+    function onError(err){
+        showNotification(null);
+    }
+    browser.tabs.query({currentWindow: true, active: true}).then(logTabs, onError);
+} 
 browser.tabs.onActivated.addListener(updateCurrentUrl);
-browser.tabs.onUpdated.addListener(updateCurrentUrl);
 
 //service is loaded into the page (getCurrentURL.js)
 const service = browser.runtime.connect({name:"updateWindowURL"});
+
+function findBetterAlternative(currentUrl) {
+    console.log('AlternativeJS: ', alternativeApps)
+    let betterAlternative
+    alternativeApps.softwaresList.forEach((next, key) => {
+        if(next.url[0] == currentUrl) {
+            betterAlternative = next.alternatives[0].url
```

Comment:
This looks good, but maybe we can improve it a bit. Considering there are multiple alternatives, would it make sense to return the alternatives array, and present them to the user?

Reply:
Yes, this is a good idea. I tried to give more alternatives than one but it didn't work as an array, but if it works then I can take your suggestion. I will try it.

Reply:
So, the idea here is that we can return the array ```betterAlternative = next.alternatives```
and then later in the code you can use ```betterAlternative[0].url``` or iterate the array to present multiple alternatives. It will involve some work, changing code in multiple locations, so it's up to you if you think it's worth making these changes now or saving it for later.